base/arrayshow.jl

```diff
@@ -480,12 +480,21 @@ function typeinfo_prefix(io::IO, X)
     # what the context already knows about the eltype of X:
     eltype_ctx = typeinfo_eltype(typeinfo)
     eltype_X = eltype(X)
-    # Types hard-coded here are those which are created by default for a given syntax
-    if eltype_X == eltype_ctx || !isempty(X) && eltype_X in (Float64, Int, Char, String)
-        ""
-    elseif print_without_params(eltype_X)
-        string(unwrap_unionall(eltype_X).name) # Print "Array" rather than "Array{T,N}"
+    if X isa AbstractDict
+        K, V = typeof(X).parameters
+        if eltype_X == eltype_ctx || !isempty(X) && isconcretetype(K) && isconcretetype(V)
+            string(typeof(X).name)
```

Comment:
Are you sure `unwrap_unionall` isn't needed as in the other branch? I don't remember why it's there.

Reply:
No I'm not sure... I still need to see an example where this make a difference, to better understand, so I preferred for now to not change the logic from what it was in the Dict code.